src/components/curator-actions/child-bounties/operations/ReassignSubCurator.svelte

```diff
@@ -0,0 +1,159 @@
+<script lang="ts">
+	import { convertFormattedDotToPlanck, isValidAddress } from '../../../../utils/polkadot';
+	import { dotApi } from '../../../../stores';
+	import { showErrorModal } from '../../../modals';
+	import type { ChildBounty } from '../../../../types/child-bounty';
+	import { isPositiveNumber } from '../../../../utils/common';
+	import Input from '../../../Input/Input.module.css';
+	import Dialog from '../../../common/Dialog.svelte';
+	import { MultiAddress } from '@polkadot-api/descriptors';
+	import { maybeTransaction, submitTransaction } from '../../../../utils/transaction';
+	import ExtendBountyLabel from '../../../ExtendBountyLabel.svelte';
+	import Fee from '../../../Fee.svelte';
+	import type { Bounty } from '../../../../types/bounty';
+	import CopyableAddress from '../../../common/CopyableAddress.svelte';
+	import { Binary } from 'polkadot-api';
+
+	export let dialog: HTMLDialogElement;
+	export let childBounty: ChildBounty;
+	export let parentBounty: Bounty;
+
+	let curatorAddress = '';
+	let curatorFee = '';
+	let extend = false;
+
+	$: transaction = maybeTransaction(() => {
+		if (!parentBounty.curator || !isValidAddress(curatorAddress) || !isPositiveNumber(curatorFee))
+			return;
+
+		const propose = $dotApi.tx.ChildBounties.propose_curator({
+			parent_bounty_id: childBounty.parentBounty,
+			child_bounty_id: childBounty.id,
+			curator: MultiAddress.Id(curatorAddress),
+			fee: convertFormattedDotToPlanck(curatorFee)
+		});
+
+		const unassign = $dotApi.tx.ChildBounties.unassign_curator({
+			child_bounty_id: childBounty.id,
+			parent_bounty_id: childBounty.parentBounty
+		});
+
+		const extendTx = $dotApi.tx.Bounties.extend_bounty_expiry({
+			bounty_id: childBounty.parentBounty,
+			remark: new Binary(new Uint8Array())
+		});
+
+		return $dotApi.tx.Utility.batch_all({
+			calls: [unassign.decodedCall, propose.decodedCall, ...(extend ? [extendTx.decodedCall] : [])]
+		});
+	});
+
+	async function submit() {
+		if (!isValidAddress(curatorAddress)) {
+			showErrorModal('Curator address is invalid.');
+			return;
+		}
+
+		if (!isPositiveNumber(curatorFee)) {
+			showErrorModal('Curator fee value is invalid.');
+			return;
+		}
+
+		if (!transaction) {
+			showErrorModal('Transaction failed.');
```

Comment:
Do you think this is a right error message? As a user of web3 tools, I’d understand it as "the transaction was submitted to blockchain and failed there". I think other dialogs report "unexpected error" at similar stages, because we really don’t expect this check to fail.